arch/arm/src/nrf52/sdc/nrf.h

```diff
@@ -0,0 +1,84 @@
+/****************************************************************************
+ * arch/arm/src/chip/nrf52_sdc.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_ARM_SRC_NRF52_SDC_NRF_H__
+#define __ARCH_ARM_SRC_NRF52_SDC_NRF_H__
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* This is the only definition we need from NRFX. We could simply do a
+ * typedef to an integer but we cannot ensure it will end up being the right
+ * size, so we replicate all values here.
```

Comment:
This does not actually solve the issue and one reason I _hate_ when enums are used on an API boundary.  The ABI is set by the compiler flags that build the library ARM for arm-none-eabi does not have hard restriction on this, but the nrf SDK uses the `-fshort-enums` flag for gcc so we can be fairly sure that is a signed short, so I would say we should just typedef this and cast.

Reply:
Ah, I didn't know you could know the build flags used for the library. I will typedef to int16_t then.